src/matrix/kaldi-gpsr.cc

```diff
@@ -226,9 +226,14 @@ double GpsrBasic(const GpsrConfig &opts, const SpMatrix<double> &H,
   Vector<double> delta_v(dim);
   Vector<double> u_new(dim);
   Vector<double> v_new(dim);
-  double objf_old, objf_new, num_zeros;
+  double objf_old = -std::numeric_limits<double>::infinity(),
+    objf_new = -std::numeric_limits<double>::infinity(),
+    num_zeros;
   bool keep_going = true;
 
+  KALDI_ASSERT(opts.max_iters_backtrak > 0 &&
```

Comment:
probably best to fix the typo in this variable name, since we are touching it.

Reply:
Sure. By the way, as far as I can tell, kaldi-gpsr is not used by anything within Kaldi. Maybe this was forgotten when one of the sgmm implementations was removed.

Reply:
Interesting.  Yes, it doesn't seem to be used.  @desh2608 would you mind making a PR to remove it?  Don't forget to change the Makefile in matrix/.

Reply:
Ok I'm on it.

Reply:
@danpovey should I remove the entire file and the associated header file as well?